### PR TITLE
Bump Maps SDK to v10.0.0-beta.13 and migrate to the new location plugin impl

### LINKED ISSUE #3952: Integrate refactored Location Plugin
@LukasPaczos commented on [Fri Jan 15 2021](https://github.com/mapbox/mapbox-navigation-android-internal/issues/78)

Refs https://github.com/mapbox/mapbox-maps-android/pull/945.

---

@LukasPaczos commented on [Tue Jan 26 2021](https://github.com/mapbox/mapbox-navigation-android-internal/issues/78#issuecomment-767697650)

Blocked by https://github.com/mapbox/mapbox-maps-android-internal/issues/215.

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

Closes #3952. This PR upgrades the Maps SDK to the latest `v10.0.0-beta.13` which reintroduces the `OnIndicatorPositionChangedListener` which means that we can rely on the new `LocationComponentPlugin` implementation instead of the legacy one.

Since the new `LocationComponentPlugin` introduces a concept of a `LocationProvider`, this PR also adds a utility class that can be paired with Nav SDK's enhanced location generators to transition the puck.

### Changelog

```
<changelog>Upgraded Maps SDK top `v10.0.0-beta.13`. Together with the update, Nav SDK switches the puck rendering dependency to the new `LocationComponentPlugin`. You can use the `NavigationLocationProvider` to feed the puck with enhanced location updates.</changelog>
```

### CODE REVIEW

**Comment by LukasPaczos** [comment]
The PR also refactors test activities to take advantage of the new location provider.

**Comment by zugaldia** [comment]
:fire:

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/location/NavigationLocationProvider.kt [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+     * @see LocationObserver
+     * @see MapMatcherResultObserver
+     * @see MapboxNavigation.registerLocationObserver
+     * @see MapboxNavigation.registerMapMatcherResultObserver
+     */
+    fun changePosition(
+        location: Location,
+        keyPoints: List<Location> = emptyList(),
+        latLngTransitionOptions: (ValueAnimator.() -> Unit)? = null,
+        bearingTransitionOptions: (ValueAnimator.() -> Unit)? = null
```
Comment (Guardiola31337) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/location/NavigationLocationProvider.kt:141]: Should we include docs (`@param`) about these?

File: test-app/src/main/java/com/mapbox/navigation/examples/core/MapboxSnapshotActivity.java [hunk 2]
```diff
@@ -305,10 +250,6 @@ private void updateLocation(List<Location> locations) {
     );
   }
 
-  private LocationPluginImpl getLocationComponent() {
-    return mapView.getPlugin(LocationPluginImpl.class);
-  }
-
   private CameraAnimationsPlugin getMapCamera() {
     return CameraAnimationsPluginImplKt.getCameraAnimationsPlugin(mapView);
```
Comment (Guardiola31337) [inline test-app/src/main/java/com/mapbox/navigation/examples/core/MapboxSnapshotActivity.java:254]: Maps should add `@file:JvmName("CameraAnimationsPluginImpl")` to make it more Java user-friendly (remove the need of `Kt`) cc @tobrun

**Review by Guardiola31337 — APPROVED** [review]
> The PR also refactors test activities to take advantage of the new location provider.

Noticed that the location always initializes in SF and moves right after to your current location (as soon it is available) 👀 

test_app_location_puck_issues

Is that expected?

Also, as seen the video, tested changing the style and the puck does not survive / blinks. Is that a downstream issue or example setup?

BTW, these are minor comments not blocking the PR 🚀

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/location/NavigationLocationProvider.kt (hunk 1 above)
Comment (LukasPaczos) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/location/NavigationLocationProvider.kt:145]: Done.

**Comment by LukasPaczos** [comment]
> Is that expected?

That's a regression, tracking in https://github.com/mapbox/mapbox-maps-android/issues/101.

**Comment by LukasPaczos** [comment]
> Noticed that the location always initializes in SF

I believe this is coming from the style definition. Maps SDK now respects the initial camera position value from the style and that's where it initializes.

**Pull request merged**